File: backend/migration-script/product.js
// const mongo = require('mongodb')
// const axios = require('axios')
// const chanceInstance = new chance()
// const Config = require("./config")

// async function sendToSearch(sProducts) {
//     await axios
//     .post(Config.ProductBulkInsertURL, sProducts)
//     .then(res => {
//       console.log(`statusCode: ${res.statusCode}`)
//     })
//     .catch(error => {
//       console.error(error.response.data)
//     })
// }

// function CtoSProduct(counter) {
//     return {
//         id : counter,
//         shop_item_id: cProduct.shop_item_id,
//         version: cProduct.version,
//         slug: cProduct.shop_item_slug,
//         name: cProduct.item_name,
//         shop_name: cProduct.shop_name,
//         shop_slug: cProduct.shop_slug,
//         brand_name: cProduct.brand_name,
//         brand_slug: cProduct.brand_slug,
//         category_name: cProduct.category_name,
//         category_slug: cProduct.category_slug,
//         color: cProduct.color,
//         color_variants: cProduct.color_variants,
//         discounted_price: cProduct.discounted_price,
//         max_price: cProduct.max_price,
//         min_price: cProduct.min_price,
//         price: cProduct.item_price,
//         product_image: cProduct.item_images ? cProduct.item_images.length ? cProduct.item_images[0]: "" : "",
//     }
// }

// (async function main() {
//     let counter = 1
//     let data = []
//     while (counter < 50000) {
//         data = [...data, CtoSProduct(counter)]
//         console.log(counter)
//         if (counter % 500 == 0) {
//             await sendToSearch(data)
//             console.log(counter, " products inserted")
//             data = []
//         }
       
//         counter += 1
//     }
//     console.log("finished bulk inserting products")
//     process.exit(0)
// })()

File: backend/search/product.go
package search

// Product defines product type
type Product struct {
	ID              int64         `json:"id" mapstructure:"id"`
	Version         int64         `json:"version" mapstructure:"version"`
	Slug            string        `json:"slug" mapstructure:"slug"`
	Name            string        `json:"name" mapstructure:"name"`
	ShopName        string        `json:"shop_name" mapstructure:"shop_name"`
	ShopSlug        string        `json:"shop_slug" mapstructure:"shop_slug"`
	ShopItemID      int64         `json:"shop_item_id,omitempty" mapstructure:"shop_item_id"`
	Price           float64       `json:"price,omitempty" mapstructure:"price"`
	DiscountedPrice float64       `json:"discounted_price,omitempty" mapstructure:"discounted_price"`
	MinPrice        float64       `json:"min_price,omitempty" mapstructure:"min_price"`
	MaxPrice        float64       `json:"max_price,omitempty" mapstructure:"max_price"`
	BrandName       string        `json:"brand_name,omitempty" mapstructure:"brand_name"`
	BrandSlug       string        `json:"brand_slug,omitempty" mapstructure:"brand_slug"`
	CategoryName    string        `json:"category_name,omitempty" mapstructure:"category_name"`
	CategorySlug    string        `json:"category_slug,omitempty" mapstructure:"category_slug"`
	ColorVariants   []string      `json:"color_variants,omitempty" mapstructure:"color_variants"`
	Color           string        `json:"color,omitempty" mapstructure:"color"`
	Ranking         float64       `json:"ranking,omitempty" mapstructure:"ranking"`
	Tags            []string      `json:"tags,omitempty" mapstructure:"tags"`
	ClickStreams    []ClickStream `json:"click_streams,omitempty"`
	Status          bool          `json:"status,omitempty" mapstructure:"status"`
	ProductImage    string        `json:"product_image,omitempty" mapstructure:"product_image"`
}

type ClickStream struct {
	//TODO: figure out later
}
